### DISCUSSION TITLE
Is there a way to free data in class Packet? (or delete the class itself?)

### DISCUSSION THREAD

**Discussion by hohoho4u4me** [ANLAB-KAIST/KENSv3#216] [Project Q&A 2. Transport Layer (TCP)] [answered]
When doing write, we should create and send lots of packet to the opponent. Therefore I'm concerned of allocated data inside packets and want to free or delete it after sending. 
However I could not find a way to free data inside it's buffer because it is private. Would there be a way to remove allocated data inside packet or directly remove data of packet(that I already sended)? 

class Packet : public Module::MessageBase {
private:
  Packet(UUID uuid, size_t maxSize);
  **std::vector<char> buffer;**

**Comment by Pusnow** [comment, marked as answer]
They are autometically freeed. You don't neet to manually free them (similar to `unique_ptr`).

**Reply by hohoho4u4me** [reply]
Thankyou!!
